tests/helpers/txparser/txtypes.py

```diff
@@ -0,0 +1,204 @@
+from io import BytesIO
+from sys import version_info
+from typing import NewType, Optional, cast
+from dataclasses import dataclass
+assert version_info.major >= 3, "Python 3 required!"
+if version_info.minor >= 8:
+    from typing import Literal
+elif version_info.minor <= 6:   # TypedDict & Literal not yet standard in 3.6
+    from typing_extensions import Literal
+
+# Types of the transaction fields, used to check fields lengths
+u8 = NewType("u8", int)  # 1-byte int
+u16 = NewType("u16", int)  # 2-byte int
+u32 = NewType("u32", int)  # 4-byte int
+u64 = NewType("u64", int)  # 8-byte int
+i64 = NewType("i64", int)  # 8-byte int, signed
+varint = NewType("varint", int)     # 1-9 bytes
+byte = NewType("byte", type(bytes(1)))          # 1 byte
+bytes2 = NewType("bytes2", type(bytes(2)))      # 2 bytes
+bytes4 = NewType("bytes4", type(bytes(4)))      # 4 bytes
+bytes8 = NewType("bytes8", type(bytes(8)))      # 8 bytes
+bytes16 = NewType("bytes16", type(bytes(16)))   # 16 bytes
+bytes32 = NewType("bytes32", type(bytes(32)))   # 32 bytes
+bytes64 = NewType("bytes64", type(bytes(64)))   # 64 bytes
+txtype = NewType("txtype", int)
+lbstr = NewType("lbstr", Literal['big', 'little'])
```

Comment:
Using a new type is not that important here. Also, I would rename the type to be something more verbose:

```py
BYTE_ORDER = Literal["big", "little"]  # literal types are conventionally uppercase
```

Reply:
I made that type instead of relying only on `str` to that a test writer is made aware through type hinting that only 'big' or 'little' are allowed as strings.
I agree with you on the verbosity of that type however: something like `byte_order` or `endianness` (since I've made all new types lowercase)

Reply:
I was talking more about the fact that it's not necessary to use `NewType`, as you can just use `Literal`.

Ok for the naming, as long as we're consistent. 👍

Reply:
Done, type `lbstr` renamed as `byteorder` and defined as `Literal['big, 'little']`